feather/server/src/packet_handlers.rs

```diff
@@ -123,10 +127,34 @@ fn handle_animation(
     Ok(())
 }
 
-fn handle_chat_message(game: &Game, player: EntityRef, packet: client::ChatMessage) -> SysResult {
-    let name = player.get::<Name>()?;
-    let message = Text::translate_with("chat.type.text", vec![name.to_string(), packet.message]);
-    game.broadcast_chat(ChatKind::PlayerChat, message);
+fn handle_chat_message(
+    game: &mut Game,
+    player_id: Entity,
+    packet: client::ChatMessage,
+) -> SysResult {
+    // Determine if chat msg is a command, unwrap won't panic because dispatcher is set on startup of server (in main)
+    let dispatcher = Rc::clone(&*game.resources.get::<Rc<RefCell<CommandDispatcher>>>()?);
+    let dispatcher = dispatcher.borrow();
+
+    match dispatcher.call(
+        game,
+        packet.message.as_str(),
+        Some(EntityId(player_id.to_bits())),
+    ) {
+        Err(_) => {
+            // Command did not run because parsing failed, or len ofmessage was longer then u32::MAX
+            let player = game.ecs.entity(player_id)?;
+            let name = player.get::<Name>()?;
+            let message =
+                Text::translate_with("chat.type.text", vec![name.to_string(), packet.message]);
+            game.broadcast_chat(ChatKind::PlayerChat, message);
```

Comment:
I don't think this is an appropriate way too handle chat messages, since regular chat messages without a prefixed `/` should not result in an error message, and messages that result in an erroneous command execution shouldn't be broadcasted either. A more appropriate way too handle this, would be to first check for a prefix `/` and then handle eventual command execution errors.

Reply:
This actually does not print a error message when a error happens. Notice how the error is never used.  Its the opposite way around. It does not print anything if a command succeeds.  So yeah, if you write "/not a actual command", then  "/not a actual command" appears in the chat.  This is like you noted not the correct behaviour.  
I am going to change this code to be clearer.